### PR TITLE
Implement Katana Runner in Dojo

### PR DESCRIPTION
**Pull request by neotheprogramist**
fixes https://github.com/neotheprogramist/dojo/issues/2

### CODE REVIEW

**Review by tarrencev — CHANGES_REQUESTED** [review]
the katana runner shouldn't need to introduce a new binary. we already build on as part of katana. the runner should be a simple rust library that calls the katana binary through the shell, supports configuring cli (can all be default for the initial impl), handles forwarding logs to stderr and tears down the instance when it goes out of scope

File: crates/katana/runner/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[package]
+name = "katana-runner"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+anyhow.workspace = true
+katana-core = { path = "../core", default-features = false }
```
Comment (tarrencev) [inline crates/katana/runner/Cargo.toml:10]: can we remove these dependencies on katana?

**Review by tarrencev — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -78,7 +78,8 @@ jobs:
           toolchain: ${{ env.RUST_VERSION }}
       - uses: Swatinem/rust-cache@v2
       - uses: arduino/setup-protoc@v2
-      - run: cargo run --bin sozo -- --manifest-path crates/dojo-core/Scarb.toml test
+      - run: curl -L https://install.dojoengine.org | bash && source /home/runner/.bashrc && echo $PATH && /home/runner/.config/.dojo/bin/dojoup
```
Comment (tarrencev) [inline .github/workflows/ci.yml:81]: perhaps it makes sense to use the dojo docker image for this job? it comes with katana installed

Comment (matzayonc) [reply .github/workflows/ci.yml:81]: That definitely makes sense.

By the way I noticed that dojoup is downloaded to `/home/runner/.config/.dojo/bin/dojoup`, not sure if it's the expected behaviour

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -26,6 +26,12 @@ jobs:
           repo-token: ${{ secrets.GITHUB_TOKEN }}
       - name: Install nextest test runner
         uses: taiki-e/install-action@nextest
+
+      - name: Build and run dev container task
+        uses: devcontainers/ci@v0.3
```
Comment (tarrencev) [inline .github/workflows/ci.yml:31]: hm can we use the prebuilt one? we might have to update it (and eventually automate it)

Comment (matzayonc) [reply .github/workflows/ci.yml:31]: sure, just trying it out for now

**Pull request merged**